tests/phpunit/CRM/RcBase/Api/CreateHeadlessTest.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test API Create class
+ *
+ * @group headless
+ */
+class CRM_RcBase_Api_CreateHeadlessTest extends TestCase implements HeadlessInterface
+{
+    /**
+     * Test contact ID
+     *
+     * @var int
+     */
+    private $testContactId;
+
+    /**
+     * The setupHeadless function runs at the start of each test case, right before
+     * the headless environment reboots.
+     *
+     * It should perform any necessary steps required for putting the database
+     * in a consistent baseline -- such as loading schema and extensions.
+     *
+     * The utility `\Civi\Test::headless()` provides a number of helper functions
+     * for managing this setup, and it includes optimizations to avoid redundant
+     * setup work.
+     *
+     * @see \Civi\Test
+     */
+    public function setUpHeadless()
+    {
+        return \Civi\Test::headless()
+            ->installMe(__DIR__)
+            ->apply();
+    }
+
+    /**
+     * Create a clean DB before running tests
+     *
+     * @throws CRM_Extension_Exception_ParseException
+     */
+    public static function setUpBeforeClass(): void
+    {
+        // Set up a clean DB
+        \Civi\Test::headless()
+            ->installMe(__DIR__)
+            ->apply(true);
+    }
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        // Create test contact
+        $contact_data = [
+            'values' => [
+                'contact_type' => 'Individual',
+                'first_name' => 'Julius',
+                'last_name' => 'Caesar',
+            ],
+        ];
+        $user = cv("api4 Contact.create '".json_encode($contact_data)."'");
+        $this->testContactId = $user[0]['id'];
+    }
+
+    /**
+     * @throws CRM_Core_Exception
+     */
+    public function testCreateContact()
+    {
+        $contact = [
+            'contact_type' => 'Individual',
+            'first_name' => 'Scipio',
+            'external_identifier' => '111',
+        ];
+
+        // Create user
+        $contact_id = CRM_RcBase_Api_Create::contact($contact);
+
+        // Get data
+        $id = cv(
+            "api4 Contact.get +s id +w external_identifier=".$contact['external_identifier']
+        );
+        $data = cv(
+            "api4 Contact.get +s contact_type,first_name,external_identifier +w external_identifier=".$contact['external_identifier']
+        );
+
+        // Check valid ID
+        $this->assertSame($id[0]['id'], $contact_id, 'Bad contact ID returned');
```

Comment:
I think the number of returned contacts could be also checked. Something like `assertEquals(1, count($id)) `.